docs/decisions/0025-stripe-event-consumption-delivery.rst

```diff
@@ -0,0 +1,108 @@
+0025 Stripe Event Consumption And Delivery
+******************************************
+
+Status
+======
+**In progress** (March 2025)
+
+Context
+=======
+
+We intend to utilize Stripe to manage subscription and payment lifecycles for the Enterprise Self-Service Purchasing
+feature. We will need a robust way to ingest events from Stripe, handle those events, and forward them to SalesForce.
+For example, when stripe emits an ``invoice.paid`` event, that needs to be communicated to SalesForce in order to
+trigger the creation/update of Customers, Opportunities, etc. Furthermore, that same event needs to be communicated to
+our edX Enterprise backend so that we can immediately followup with the customer to confirm that we have begun to
+provision their product.
```

Comment:
[clarification] Is it this proposed event consumption/delivery system that will be communicating to the edX Enterprise backend? Or are we relying on SalesForce's handling of the forwarded events to trigger a request to the Provisioning API to communicate with the edX Enterprise backend?

Reply:
Both!  State-altering actions need to be handled by SalesForce calling our provisioning API.  However, we also may need to perform non-state-altering actions (e.g. product pending emails) and actions which should be triggered by events that we won't even send to SalesForce (e.g. subscription canceled emails), direct and immediate ingestion/handling of events into the edX backend would be required.

My proposed SSP architecture isolates the domain of recurring payments to Stripe + edX backend, and SalesForce would not be aware that the invoices it receives are part of a schedule or attached to a Stripe subscription. Hence, not all events would be delivered to SalesForce, only the invoice.paid events.

Reply:
Got it, thanks for the clarification. This makes sense :)